tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_docker.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+"""
+SPDX-License-Identifier: BSD-2-Clause-Patent
+SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+This code is subject to the terms of the BSD+Patent license.
+See LICENSE file for more details.
+"""
+
+import logging
+import os
+import random
+import subprocess
+import sys
+import time
+
+from boardfarm.devices import debian
+from environment import ALEntityDocker, _get_bridge_interface
+
+rootdir = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../../..'))
+
+logging.basicConfig(stream=sys.stdout, format='%(message)s')
+logger = logging.getLogger('bft')
+logger.setLevel(logging.INFO)  # DEBUG, INFO, WARNING, ERROR, CRITICAL
+
+
+class PrplMeshDocker(debian.DebianBox):
+    '''
+    Dockerized prplMesh device
+    '''
+    model = ("prplmesh_docker")
+
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+        config = kwargs["config"] if kwargs.get("config", None) else kwargs
+
+        # String buffer which stores all calls to device's consoles
+        self.log_calls = ""
+
+        # String buffer with device's console log
+        self.log = ""
+
+        # List of device's consoles test can interact with
+        self.consoles = [self]
+
+        # Getting unic ID to distinguish devices and network they belong to
+        self.unique_id = os.getenv("SUDO_USER", os.getenv("USER", ""))
+
+        self.name = "-".join((config.get("name", "prplmesh_docker"), self.unique_id))
+        self.role = config.get("role", "agent")
+        self.cleanup_cmd = config.get("cleanup_cmd", None)
+        self.conn_cmd = config.get("conn_cmd", None)
+        self.mac = config.get("mac", self._get_rand_mac(oct_amount=4))
+        self.use_tcpdump = config.get("use_tcpdump", False)
+        self.docker_network = config.get("docker_network",
+                                         "prplMesh-net-{}".format(self.unique_id))
+
+        if self.cleanup_cmd:
+            self._run_shell_cmd(self.cleanup_cmd)
+
+        docker_run_cmd = os.path.join(rootdir, "tools", "docker", "run.sh") \
+            + " --verbose --detach --force --name {}".format(self.name) \
+            + " --mac {}".format(self.mac) \
+            + " --network {}".format(self.docker_network)
+
+        if self.role == "both":
+            # Spawn dockerized controller + agent in single device
+            self._run_shell_cmd(docker_run_cmd + " start-controller-agent")
+            time.sleep(config.get("delay", 5))
+            self.al_entity = ALEntityDocker(self.name, is_controller=True)
```

Comment:
You should create two entities here: one for the agent and one for the controller.
```suggestion
            self.agent_entity = ALEntityDocker(self.name, is_controller=False)
            self.controller_entity = ALEntityDocker(self.name, is_controller=True)
```

For the other cases, we leave either agent_entity or controller_entity None.